sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala

```diff
@@ -356,26 +380,35 @@ object InMemoryFileIndex extends Logging {
           bulkListLeafFiles(
             dirs.map(_.getPath),
             hadoopConf,
-            filter,
+            filters,
             session,
-            areRootPaths = false
+            areRootPaths = false,
+            parameters = parameters
           ).flatMap(_._2)
         case _ =>
           dirs.flatMap { dir =>
             listLeafFiles(
               dir.getPath,
               hadoopConf,
-              filter,
+              filters,
               sessionOpt,
               ignoreMissingFiles = ignoreMissingFiles,
               ignoreLocality = ignoreLocality,
-              isRootPath = false)
+              isRootPath = false,
+              parameters = parameters)
           }
       }
+      val fileFilters = (filters ++ FileModifiedDateOption.accept(
+        parameters,
+        sessionOpt.get,
+        hadoopConf))
+          .filter(filter => filter !=null)
+
       val allFiles = topLevelFiles ++ nestedFiles
-      if (filter != null) allFiles.filter(f => filter.accept(f.getPath)) else allFiles
```

Comment:
Once we added the ability to use classes extending from PathFilter, I realized we would only allow one filter to be applied at anytime.  Given the number of consumers calling InMemoryFileIndex methods, this seemed unideal, so I extended this to use a parallel array supporting multiple filters.